docs/how-to/integrate-external-libraries.md

```diff
@@ -0,0 +1,252 @@
+# How-To: Integrate a Third-Party Library
+
+> [!IMPORTANT]
+> For this How-To guide and many other advanced F´ CMake use cases, it is recommended to have a basic understanding of CMake. If you are unfamiliar with CMake, please refer to the [CMake documentation](https://cmake.org/getting-started/). They provide a [great tutorial to get you started](https://cmake.org/cmake/help/latest/guide/tutorial/index.html).
+
+It is common practice to use third-party libraries in F´ projects. CMake provides multiple ways to integrate external libraries into a project, and the F´ build system facilitates it even further. This document will cover four different approaches to integrate external libraries into your F´ project design and build process. You should also refer to each library's documentation for any specific requirements or recommendations for integration, as there are many different ways to build and use libraries in CMake.
+
+> [!NOTE]
+> The source code for the examples shown in this guide can be found in our examples repo here: [https://github.com/nasa/fprime-examples/tree/devel/FlightExamples/ExternalLibs](https://github.com/nasa/fprime-examples/tree/devel/FlightExamples/ExternalLibs)
+
+## Common considerations
+
+CMake provides a [Using Dependencies Guide](https://cmake.org/cmake/help/v3.31/guide/using-dependencies/index.html) which we recommend reading for a deeper understanding of how to integrate dependencies into your CMake projects.
+
+When integrating a library into your F´ project, there are considerations based on how the library is built and what it provides.
+
+- **Source or Pre-compiled**: Do you have the source code for the library, or is it pre-compiled (installed on your system, or `.a / .so` files available)?
+- **Build System** _(only applies if you have the source)_: Does the library use CMake, or another build system?
+- **CMake integration**: Does the library provide CMake configuration files such as `<libName>Config.cmake`? (refer to the [CMake documentation](https://cmake.org/cmake/help/latest/guide/using-dependencies/index.html) for more information)
+
+The following summarizes recommendation on which approach to use based on these considerations:
+
+1. **Pre-compiled**:
```

Comment:
Perhaps this should be a flow chart?

```mermaid
flowchart TD
    Library-->Source{Source?}
    Source-->|Yes| CMakeSourceSupport{Supports CMake?}
        CMakeSourceSupport-->|Yes|IsSubmodule{Is submodule/folder?}
            IsSubmodule-->|Yes| AddSubdirectory[Approach 2: add_subdirectory]
            IsSubmodule-->|No| FetchContent[Approach 2: FetchContent]
        CMakeSourceSupport-->|No|ExternalProject_Add[Approach 4: ExternalProject_Add]
    Source-->|No| CMakePackageSupport{Supports CMake?}
        CMakePackageSupport-->|Yes| find_package[Approach 1: find_package]
        CMakePackageSupport-->|No| MOD_DEPS[Approach 2: MOD_DEPS]
```

Reply:
You should audit the above for clarity and correctness.  Approaches should likely be enumerated left to right.